### PR TITLE
[crypto] Replace #5099 with a better fix

### PR DESCRIPTION
**Pull request by Damian-Nordic**
#### Problem
The quick-fix I did in #5099 turned out to be harmful for controllers which don't use kTestControllerNodeId to identify the controller node. 

 #### Summary of Changes
Provide a more flexible fix which hopefully works correctly for all controllers apps.

I tested it with the newly added Python CHIP Controller flag which allows to set an arbitrary controller node ID.

### CODE REVIEW

**Comment by Damian-Nordic** [comment]
@sagar-apple could you check if the fix helps with the problem in iOS CHIPTool we were talking about?

**Review by sagar-apple — APPROVED** [review]
Confirmed this fixes the iOS controller as well. Thanks Damian!

File: src/transport/RendezvousSession.cpp [hunk 1]
```diff
@@ -433,20 +432,24 @@ void RendezvousSession::InitPairingSessionHandle()
...
-        mPairingSessionHandle = nullptr;
+        // Reset the transport and peer address in the active secure channel
+        // This will allow the regular transport (e.g. UDP) to take over the existing secure channel
+        state->SetTransport(nullptr);
+        state->SetPeerAddress(PeerAddress{});
+
+        // When the remote node ID is not specified in the initial rendezvous parameters, the connection state
+        // is created with undefined peer node ID. Update it now.
+        if (state->GetPeerNodeId() == kUndefinedNodeId)
+            state->SetPeerNodeId(mParams.GetRemoteNodeId().ValueOr(kUndefinedNodeId));
```
Comment (andy31415) [inline src/transport/RendezvousSession.cpp:448]: Maybe we should log if mParams.GetRemoteNodeId() is not set.
This kind of errors are very hard to track.

**Review by andy31415 — APPROVED** [review]

**Comment by andy31415** [comment]
Marking as hotfix: provision broken on non-default node ids seems a problem for TE1.

Please address any review comments in a followup PR.

**Pull request merged**